components/src/dynamo/vllm/handlers.py

```
@@ -97,6 +98,20 @@ def build_sampling_params(
+        # StructuredOutputsConfig.backend (often "auto"). Do NOT set request-level backend
+        # here, because vLLM rejects per-request backend selection when the engine is
+        # initialized with a different backend (see vLLM _validate_structured_output).
+        #
+        # If the engine backend is a concrete value (not "auto"), we can optionally set it
+        # for older vLLM variants that rely on StructuredOutputsParams._backend. For vLLM
+        # v0.12+, leaving it unset is fine and preferred.
+        if structured_outputs_backend and structured_outputs_backend != "auto":
+            try:
+                sampling_params.structured_outputs._backend = structured_outputs_backend  # type: ignore[attr-defined]
+            except Exception:
+                pass
```

Ruff (0.14.10)

`try`-`except`-`pass` detected, consider logging the exception

(S110)